src/urllib3/response.py

```diff
@@ -1020,10 +1020,23 @@ def stream(
             yield from self.read_chunked(amt, decode_content=decode_content)
         else:
             while not is_fp_closed(self._fp) or len(self._decoded_buffer) > 0:
-                data = self.read(amt=amt, decode_content=decode_content)
-
+                if amt is None:
+                    data = self.read1(amt=amt, decode_content=decode_content)
+                else:
+                    data = self.read(amt=amt, decode_content=decode_content)
                 if data:
                     yield data
+                else:
+                    # http.client.HTTPResponse will often close this for us and
```

Comment:
This isn't the right place for this to go. This comment makes me very apprehensive about merging this and also makes me worry about the correctness of the `read1` PR.